kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableConfig.java

```diff
@@ -80,6 +120,15 @@ public class TableConfig<T> {
                 TOMBSTONE_RETENTION.getKey().toLowerCase(Locale.ROOT), TOMBSTONE_RETENTION);
         validProperties.put(
                 CHECKPOINT_INTERVAL.getKey().toLowerCase(Locale.ROOT), CHECKPOINT_INTERVAL);
+        validProperties.put(
+                IN_COMMIT_TIMESTAMPS_ENABLED.getKey().toLowerCase(Locale.ROOT),
+                IN_COMMIT_TIMESTAMPS_ENABLED);
+        validProperties.put(
+                IN_COMMIT_TIMESTAMP_ENABLEMENT_VERSION.getKey().toLowerCase(Locale.ROOT),
+                IN_COMMIT_TIMESTAMP_ENABLEMENT_VERSION);
+        validProperties.put(
+                IN_COMMIT_TIMESTAMP_ENABLEMENT_TIMESTAMP.getKey().toLowerCase(Locale.ROOT),
+                IN_COMMIT_TIMESTAMP_ENABLEMENT_TIMESTAMP);
```

Comment:
you can just add a static utility method:

```
private void addConfig(Map<String, TableConfig> configs, TableConfig config) {
    configs.put(config.getKey().toLowerCase(Locale.ROOT), config);
}
```

add just call `addConfig(validProperties, IN_COMMIT_TIMESTAMP_ENABLEMENT_TIMESTAMP)`